### DISCUSSION TITLE
Cleanup dirty data

### DISCUSSION THREAD

**Discussion by bdnstn** [klaasnicolaas/home-assistant-glow#237] [Q&A] [answered]
As you can see below, my daily energy consumption is around 20kWh. On the 10th and 11th of October I was playing with the mount of my Glow and got a bunch of extraneous daily readings of over 600 kWh.  

![image](https://user-images.githubusercontent.com/24848173/204684256-5b6e8032-8534-4790-9bc3-5f28ea92670e.png)

These dirty records make my long-term energy history very inaccurate.

Running an SQLite query for the October 10th, I can see the dirty record:

![image](https://user-images.githubusercontent.com/24848173/204688114-12b020ef-8ff5-4e42-b467-7e614a4cbfa7.png)

It seems to me it should be possible to just delete the dirty record, but there are several other `statistic` record types that probably also need to be adjusted in order not to corrupt my database.

![image](https://user-images.githubusercontent.com/24848173/204689533-86ed03f0-fac8-46f4-8eca-e0693802b47d.png)

After further reading and digging in the documentation, it looks like the `statistics` are possibly generated by the `Recorder` integration from data in the `states` table, which in the case of GLOW has `daily` accumulator records. So, I should probably delete those too, or at least edit in reasonable estimated values. Then there is the question of how to correct all the following `stastics.%_total_energy` and `statistics.%_total_energy_cost` records. It is looking complex.

From reading other discussions here it sounds like others have similar issues with data that should be removed. The question is, how to do it safely, without risk of corruption.

**Comment by klaasnicolaas** [comment, marked as answer]
You can solve these kinds of problems in the long-term statistics in developer tools -> statistics tab.

[![Open your Home Assistant instance and show your statistics developer tools.](https://my.home-assistant.io/badges/developer_statistics.svg)](https://my.home-assistant.io/redirect/developer_statistics/)

**Reply by bdnstn** [reply]
Really? I can't see anything there that will allow me to remove specific records, only removing whole datasets for entities which have no state.

Thanks for this project by the way, I love the data it gives me about my house.

**Comment by bdnstn** [comment]
OK, my bad. I didn't look far enough in the Developer Tools | Statistics.

![image](https://user-images.githubusercontent.com/24848173/204992563-1c331ac9-2d81-469e-9600-e5eea17dd2a4.png)

You have to click on the little ramp symbol on the far right.
